mcs/class/corlib/System.Security.Principal/SecurityIdentifier.cs

```diff
@@ -281,42 +281,46 @@ public override IdentityReference Translate (Type targetType)
 		internal string GetSddlForm()
 		{
 			string sidString = Value;
-			
+
 			WellKnownAccount acct = WellKnownAccount.LookupBySid(sidString);
 			if(acct == null || acct.SddlForm == null)
 				return sidString;
-			
+
 			return acct.SddlForm;
 		}
 
 		internal static SecurityIdentifier ParseSddlForm(string sddlForm, ref int pos)
 		{
 			if (sddlForm.Length - pos < 2)
 				throw new ArgumentException("Invalid SDDL string.", "sddlForm");
-			
+
 			string sid;
 			int len;
-			
+
 			string prefix = sddlForm.Substring(pos, 2).ToUpperInvariant();
 			if (prefix == "S-")
 			{
 				// Looks like a SID, try to parse it.
 				int endPos = pos;
-				
+
 				char ch = Char.ToUpperInvariant(sddlForm[endPos]);
 				while (ch == 'S' || ch == '-' || ch == 'X'
 				       || (ch >= '0' && ch <= '9') || (ch >= 'A' && ch <= 'F')) {
 					++endPos;
 					ch = Char.ToUpperInvariant(sddlForm[endPos]);
 				}
-				
+
+				if (ch == ':' && sddlForm[endPos - 1] == 'D') {
```

Comment:
GUID will be in lowercase.

SDDL has the format:
`O:{ SID }G:{ SID }D:{ SID }S:{ SID }`
The problem is when the `G:{ SID }D:` section is parsed, the `D` is accidentally consumed because GUID might also have a lower case `d`

Detect if we ended `ch` on `:` and check if the previous character is upper case `D`, if it is, rewind 1 character